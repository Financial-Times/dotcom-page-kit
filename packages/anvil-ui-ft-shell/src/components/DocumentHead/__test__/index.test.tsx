import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../'

describe('anvil-ui-ft-shell/src/components/DocumentHead', () => {
  // TODO: better tests
  it('renders the document head', () => {
    const props = {
      description: 'Website description.',
      enableJsonLD: true,
      enableOpenGraph: true,
      facebookDescription: 'Facebook description.',
      facebookHeadline: 'Facebook headline',
      facebookImage: 'facebook.jpg',
      facebookPage: 'facebook-page-id',
      mainImage: 'main-image.jpg',
      metadata: { '@type': 'article', headline: 'JSON+LD headline' },
      pageTitle: 'Page title',
      siteTitle: 'Website title!',
      stylesheets: [],
      twitterDescription: 'Twitter description.',
      twitterHeadline: 'Twitter headline',
      twitterImage: 'twitter.jpg',
      twitterSite: '@twitter_page',
      url: 'https://my.site'
    }

    const tree = renderer.create(<Subject {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
