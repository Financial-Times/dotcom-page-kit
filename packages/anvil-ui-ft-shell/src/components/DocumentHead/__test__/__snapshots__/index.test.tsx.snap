// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anvil-ui-ft-shell/src/components/DocumentHead renders the document head 1`] = `
Array [
  <meta
    charSet="utf-8"
  />,
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />,
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />,
  <title>
    Page title | Website title!
  </title>,
  <meta
    content="Website description."
    name="description"
  />,
  <link
    href="https://my.site"
    rel="canonical"
  />,
  <link
    href="https://spoor-api.ft.com"
    rel="preconnect"
  />,
  <link
    crossorigin="use-credentials"
    href="https://session-next.ft.com"
    rel="preconnect"
  />,
  <link
    href="https://ads-api.ft.com"
    rel="preconnect"
  />,
  <link
    href="https://www.googletagservices.com"
    rel="preconnect"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "background-color: #fff1e5;",
      }
    }
  />,
  <meta
    content="index,follow"
    name="robots"
  />,
  <meta
    content="4-t8sFaPvpO5FH_Gnw1dkM28CQepjzo8UjjAkdDflTw"
    name="google-site-verification"
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "{\\"@type\\":\\"article\\",\\"headline\\":\\"JSON+LD headline\\"}",
      }
    }
    type="application/ld+json"
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "{\\"@context\\":\\"http://schema.org\\",\\"@type\\":\\"WebSite\\",\\"name\\":\\"Financial Times\\",\\"alternateName\\":\\"FT.com\\",\\"url\\":\\"http://www.ft.com\\"}",
      }
    }
    type="application/ld+json"
  />,
  <meta
    content="en_GB"
    property="og:locale"
  />,
  <meta
    content="Website title!"
    property="og:site_name"
  />,
  <meta
    content="Twitter headline"
    name="twitter:title"
  />,
  <meta
    content="Facebook headline"
    property="og:title"
  />,
  <meta
    content="Twitter description."
    name="twitter:description"
  />,
  <meta
    content="Facebook description."
    property="og:description"
  />,
  <meta
    content="https://my.site"
    name="twitter:url"
  />,
  <meta
    content="https://my.site"
    property="og:url"
  />,
  <meta
    content="twitter.jpg"
    name="twitter:image"
  />,
  <meta
    content="facebook.jpg"
    property="og:image"
  />,
  <meta
    content="facebook-page-id"
    property="fb:pages"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="@twitter_page"
    name="twitter:site"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=32&height=32&format=png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=194&height=194&format=png"
    rel="icon"
    sizes="194x194"
    type="image/png"
  />,
  <link
    href="https://www.ft.com/__origami/service/image/v2/images/raw/ftlogo-v1%3Abrand-ft-logo-square-coloured?source=update-logos&width=180&height=180&format=png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
]
`;
